config/routes.rb

```diff
@@ -0,0 +1,30 @@
+Rails.application.routes.draw do
+  root 'users#index'
+
+  resources :markets, except: [:destroy] do
+    resources :vendors, except: [:index] do
+      resources :sales, only: [:create, :new]
+      resources :products, except: [:show]
+    end
+  end
+
+  get  "/all_markets"        => "users#show"
+  get  "/all_markets/:id"    => "users#market_details"
+
+  # Market index page sorts
+  get "/markets_/name_sort"      => "markets#name_sort"
+  get "/markets_/city_sort"      => "markets#city_sort"
+  get "/markets_/state_sort"     => "markets#state_sort"
+  get "/all_markets_/name_sort"  => "users#name_sort"
```

Comment:
When you refactor to remove users, then you only need the markets routes.  I'm confused here why you used markets_ rather than just markets for the sort routes?  It works without the _ for my version.